### PR TITLE
Todo App backend

### PR DESCRIPTION
**Pull request by chayanikabhandary**
## What
- Basic app setup
- Base API
- Migration
- Model

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from app import db
+
+class Task(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    task_description = db.Column(db.String(64), index=True)
+    due_date = db.Column(db.Date, nullable= False)
+    created_by = db.Column(db.String(128), nullable= False)
+    status = db.Column(db.Boolean)
+    deleted_by = db.Column(db.String(128))
+    created_at = db.Column(db.Text, nullable= False)
```
Comment (chayanikabhandary) [inline app/models.py]: Use DateTime instead of Text

File: app/models.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from app import db
+
+class Task(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    task_description = db.Column(db.String(64), index=True)
```
Comment (chayanikabhandary) [inline app/models.py]: Two solutions:
- Keep brief and detailed descriptions
- Make this the detailed description

File: app/models.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    due_date = db.Column(db.Date, nullable= False)
+    created_by = db.Column(db.String(128), nullable= False)
```
Comment (chayanikabhandary) [inline app/models.py]: Change to user id in future.

File: app/models.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    status = db.Column(db.Boolean)
```
Comment (chayanikabhandary) [inline app/models.py]: Make changes for supporting multiple statuses.

File: app/routes.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from app import app
+from flask import render_template
+from app.forms import DataForm
+
+@app.route('/')
+@app.route('/login', methods=['GET','POST'])
+def login():
+    form= DataForm()
+    if form.validate_on_submit():
+    	return redirect('/index')
```
Comment (chayanikabhandary) [inline app/routes.py]: use url_for

File: app/forms.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from flask_wtf import FlaskForm
+from wtforms import StringField, DateField, BooleanField, SubmitField, validators
+from wtforms.validators import DataRequired
+
+class DataForm(FlaskForm):
```
Comment (ghost) [inline app/forms.py]: Rename to CreateTaskForm

File: app/forms.py [hunk 7]
```diff
@@ -1,11 +1,37 @@
...
-from wtforms import StringField, DateField, BooleanField, SubmitField, validators
-from wtforms.validators import DataRequired
+from wtforms import StringField, DateTimeField, BooleanField, SubmitField, PasswordField
+from wtforms.validators import ValidationError, DataRequired, Email, EqualTo
+from app.models import User
 
 class DataForm(FlaskForm):
     task_description = StringField('Task Description: ', validators=[DataRequired()])
-    due_date = DateField('Due Date', validators=[DataRequired()], format= '%m/%d/%Y')
+    due_date = DateTimeField('Due Date', validators=[DataRequired()], format= '%Y-%m-%d %H:%M:%S')
```
Comment (dhruv-aggarwal) [inline app/forms.py]: This stores timestamp instead of date.
Rename accordingly.

File: app/forms.py [hunk 8]
```diff
@@ -1,11 +1,37 @@
...
     status = BooleanField('Status')
     created_by = StringField('Created By: ', validators=[DataRequired()])
-    created_at = DateField('Created At', validators=[DataRequired()])
-    done = SubmitField('Done')
\ No newline at end of file
+    created_at = DateTimeField('Created At: ', validators=[DataRequired()], format= '%Y-%m-%d %H:%M:%S')
```
Comment (dhruv-aggarwal) [inline app/forms.py]: Format can be moved to a constant

File: app/models.py [hunk 9]
```diff
@@ -1,16 +1,52 @@
...
-    task_description = db.Column(db.String(64), index=True)
-    due_date = db.Column(db.Date, nullable= False)
+    task_description = db.Column(db.String(256), index=True)
+    due_date = db.Column(db.DateTime, nullable= True)
     created_by = db.Column(db.String(128), nullable= False)
     status = db.Column(db.Boolean)
-    deleted_by = db.Column(db.String(128))
-    created_at = db.Column(db.Text, nullable= False)
-    deleted_at = db.Column(db.Text, nullable= True)
+    deleted_by = db.Column(db.String(128), nullable= True)
```
Comment (dhruv-aggarwal) [inline app/models.py]: Should be an integer

File: app/models.py [hunk 10]
```diff
@@ -1,16 +1,52 @@
...
+    created_at = db.Column(db.DateTime, nullable= True)
```
Comment (dhruv-aggarwal) [inline app/models.py]: check how to set default values in models and use the default value as the current time here

File: app/models.py [hunk 11]
```diff
@@ -1,16 +1,52 @@
...
+    def set_password(self, password):
+        self.password_hash = generate_password_hash(password)
+
+    def check_password(self, password):
+        return check_password_hash(self.password_hash, password)
+
+
+@login.user_loader
+def load_user(id):
+    return User.query.get(int(id))
```
Comment (dhruv-aggarwal) [inline app/models.py:51]: Handle exception in case the id is not an integer

File: app/templates/base.html [hunk 12]
```diff
@@ -7,7 +7,8 @@
       {% endif %}
     </head>
     <body>
-        <div>ToDo: <a href="/index">Home</a></div>
+        <div>ToDo: <a href="/home">Home</a> <a href='/login'>Login</a>  <a href="/logout">Logout</a></div>
```
Comment (dhruv-aggarwal) [inline app/templates/base.html]: If you are going to use React, then do not spend a lot of time on jinja templating..
Make sure that the APIs are working first.

**Pull request merged**